creator-node/src/middlewares.js

```diff
@@ -135,18 +138,28 @@ async function ensurePrimaryMiddleware(req, res, next) {
 
   /**
    * Fetch current user replicaSetSpIDs
-   * Will throw error if selfSpID is not user primary
    */
   let replicaSetSpIDs
   try {
-    replicaSetSpIDs = await getReplicaSetSpIDs({
-      serviceRegistry,
-      logger: req.logger,
+    replicaSetSpIDs = await getReplicaSetSpIdsByUserId({
+      libs: serviceRegistry.libs,
+      userId,
       blockNumber: req.body.blockNumber,
       ensurePrimary: true,
-      selfSpID,
-      userId
+      selfSpId: selfSpID,
+      parentLogger: req.logger
     })
+
+    // Error if returned primary spID does not match self spID
+    if (replicaSetSpIDs.primaryId !== selfSpID) {
+      return sendResponse(
+        req,
+        res,
+        errorResponseServerError(
```

Comment:
intentionally changed from `errorResponseUnauthorized`?

Reply:
yes because previously the ensurePrimary made the function throw and the catch would do errorResponseServerError. I'm changing it back now tho since it really should be unauthorized